### PR TITLE
refactor: keep all repository data

### PR DESCRIPTION
**Pull request by sileht**
This change keeps the repository data provided by GitHub.

This will be used by the new timed conditions to be able to send a
refresh event.

### CODE REVIEW

File: mergify_engine/context.py [hunk 1]
```diff
@@ -208,7 +219,7 @@ class Repository(object):
     @property
     def base_url(self) -> str:
         """The URL prefix to make GitHub request."""
-        return f"/repos/{self.installation.owner_login}/{self.name}"
+        return f"/repos/{self.installation.owner_login}/{self.repo['name']}"
```
Comment (jd) [inline mergify_engine/context.py:222]: Wouldn't be better to return `self.repo["url"]` at this point?

File: mergify_engine/tests/functional/actions/test_review.py [hunk 2]
```diff
@@ -47,9 +47,10 @@ async def test_review(self):
 
         p, _ = await self.create_pr()
         await self.run_engine()
-
         await self.wait_for("pull_request_review", {}),
+
         await self.run_engine()
+        await self.wait_for("pull_request_review", {}),
```
Comment (jd) [inline mergify_engine/tests/functional/actions/test_review.py:53]: sound unrelated, maybe in a different PR? 😁

**Review by jd — APPROVED** [review]

File: mergify_engine/tests/functional/actions/test_review.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/tests/functional/actions/test_review.py:53]: record was failing sometime due to this.

Comment (jd) [reply mergify_engine/tests/functional/actions/test_review.py:53]: I understood that, which is exactly my point. This should be fixed in a different PR. 😉

**Pull request merged**